src/platform/plugins/shared/data_view_field_editor/public/components/preview/field_list/field_list.tsx

```diff
@@ -201,9 +202,31 @@ export const PreviewFieldList: React.FC<Props> = ({ height, clearSearch, searchV
       </div>
     );
 
+  const toggleIsPinnedItem = useCallback(
+    (fieldName: string, keyboardEvent: { isKeyboardEvent: boolean; buttonId: string }) => {
+      controller.togglePinnedField(fieldName);
+
+      if (!keyboardEvent.isKeyboardEvent) return;
+      const newIndex = getPositionAfterToggling(fieldName, pinnedFields, fieldList);
+      // If the field is currently pinned and it goes over the limit of the fields to show we need to show all of them
+      if (newIndex > INITIAL_MAX_NUMBER_OF_FIELDS && !showAllFields) toggleShowAllFields();
```

Comment:
```suggestion
      if (newIndex >= INITIAL_MAX_NUMBER_OF_FIELDS && !showAllFields) toggleShowAllFields();
```
Looks like we might have an off by one error here where the list doesn't expand when unpinning the item right after the cutoff:

https://github.com/user-attachments/assets/406aaba2-273e-4c5b-ba0c-5bb4887e53fb